### ISSUE TITLE
undefined is not an object (evaluating 'callers')

### ISSUE THREAD

**Issue by diavrank** [pinojs/pino#1802]
Hi, maybe this is a bug with Bun.js (1.0.0), but I got an error from this package pino when I run my Nestjs app.

```shell
[Nest] 42322   - 09/09/2023, 12:35:32 AM   [ExceptionHandler] undefined is not an object (evaluating 'callers') +40ms
TypeError: undefined is not an object (evaluating 'callers')
    at fixTarget (/Users/myuser/api/node_modules/pino/lib/transport.js:129:10)
    at transport (/Users/myuser/api/node_modules/pino/lib/transport.js:115:30)
    at normalizeArgs (/Users/myuser/api/node_modules/pino/lib/tools.js:311:76)
    at pino (/Users/myuser/api/node_modules/pino/pino.js:88:4)
    at new PinoLogger (/Users/myuser/api/node_modules/nestjs-pino/PinoLogger.js:47:44)
    at <anonymous> (/Users/myuser/api/node_modules/@nestjs/core/injector/injector.js:331:30)
    at instantiateClass (/Users/myuser/api/node_modules/@nestjs/core/injector/injector.js:319:26)
    at <anonymous> (/Users/myuser/api/node_modules/@nestjs/core/injector/injector.js:49:17)
```

So, I'm not sure if this can be fixed from your side?

References:

https://github.com/oven-sh/bun/issues/4280#issuecomment-1712434399

**Comment by mcollina** [comment]
Ultimately Bun does not implement all Node.js APIs, it's actually _not_ 100% compatible (despite their claims). Please refer to the Bun [Discord](https://bun.sh/discord) or [Issue Tracker](https://github.com/oven-sh/bun) for such questions.

**Comment by pradeep-mishra** [comment]
this issue is because, pino is using Error stack to resolve the node_modules and filepath and i guess this is not supported in Bun.

i have updated pino lib/caller.js to make it work

here is the code for it

```js
'use strict'

function noOpPrepareStackTrace (_, stack) {
  return stack
}

module.exports = function getCallers () {
  const originalPrepare = Error.prepareStackTrace
  Error.prepareStackTrace = noOpPrepareStackTrace
  let stack = new Error().stack
  Error.prepareStackTrace = originalPrepare

  if (!Array.isArray(stack)) {
    let stackStrs = Array.from(stack.matchAll(/\((.+)\)/gm))
    stackStrs = stackStrs.map((match) => {
      match[1] = match[1].trim();
      return match[1].split(':')[0];
    });
    stack = stackStrs;
    //return undefined
  }

  const entries = stack.slice(2)

  const fileNames = []

  for (const entry of entries) {
    if (!entry) {
      continue
    }
    fileNames.push(entry?.getFileName?.() ?? entry)
  }
  return fileNames
}


```

**Comment by pradeep-mishra** [comment]
i can raise pull request if this works

**Comment by christophemarois** [comment]
As an update for interested future readers, this will apparently be fixed in bun `1.0.3` https://github.com/oven-sh/bun/issues/4280#issuecomment-1727666887
